client/client_update.go

```diff
@@ -86,7 +86,7 @@ func (u *UpdateClient) getUpdateInfo(api ApiRequester, process RequestProcessing
 	defer r.Body.Close()
 
 	if r.StatusCode != http.StatusOK && r.StatusCode != http.StatusNoContent {
-		if r.StatusCode == http.StatusNotFound {
+		if r.StatusCode == http.StatusNotFound || r.StatusCode == http.StatusMethodNotAllowed {
```

Comment:
This commit makes me uneasy. Don't get me wrong, this is correct, but my concern earlier about only falling back on certain error codes is already being slowly confirmed. Maybe we should just fall back on at least all of the 4xx error codes. The risk of the Open Source version breaking is really high here... What do you think?

Reply:
Most definitely! What I thought about initially. But since you wanted 404 explicitly I figured this was safe. I'll let the tests run this time though, so I don't have to wait an entire new run.

Reply:
How about just falling back on all codes not 200, just to be sure?

Reply:
I mean, `200` and `204` is handled on the above line already right? So let's just always fall back in any other case.